### PR TITLE
Migrate to leaflet 1.0.3

### PR DESCRIPTION
**Pull request by donflopez**

### CODE REVIEW

**Comment by donflopez** [comment]
Necessary for https://github.com/CartoDB/cartodb.js/issues/1503

**Comment by javisantana** [comment]
I'm not sure but I think they have not a full screen canvas layer in the new leaflet that could replace our canvas layer.

**Comment by alonsogarciapablo** [comment]
👍

**Comment by rochoa** [comment]
Should we use https://github.com/CartoDB/Leaflet.CanvasLayer as a real dependency instead of copy-pasting it?

File: lib/torque/leaflet/canvas_layer.js [hunk 1]
```diff
@@ -60,11 +60,11 @@ L.CanvasLayer = L.Class.extend({
     return canvas;
   },
 
-  onAdd: function (map) {
-    this._map = map;
+  _layerAdd: function (e) {
```
Comment (alonsogarciapablo) [inline lib/torque/leaflet/canvas_layer.js]: Why can't we use `onAdd` here? `_layerAdd` is a private method that might change in the future, so I'd try to find a better solution for this.

There's also the `onRemove` in this file. We should make sure it's working properly.

Comment (donflopez) [inline lib/torque/leaflet/canvas_layer.js]: Yes, I tried it before... but It was failing, IDK why. But now I've tried again and it works... maybe some dep fail.

I'll leave it how it was.

**Pull request merged**